### ISSUE TITLE
compatibility with caffe

### ISSUE THREAD

**Issue by Wronskia** [tidsp/caffe-jacinto#8]
Hello manu,

I tried to use a snapshot of a sparsified model from caffe-jacinto and to load it in caffe. However, I get an error stating that there is not enough weights and after checking the size of the 50% sparsified model I found that its half size of the original caffemodel. Would it be possible to have compatibility?

Thanks,
Best

**Comment by mathmanu** [comment]
Hi,

I am not sure which model. Can you please share the path of the original and the sparse model.

Thanks,

**Comment by Wronskia** [comment]
Hi,

Here is the sparse caffemodel : https://drive.google.com/file/d/0B64fJzg3TzZwSm9HeHNlT0hXSEk/view
Actually, I retrained googlenet with a subsample of imagenet with sparsity : you can download the original here : https://github.com/BVLC/caffe/tree/master/models/bvlc_googlenet

thanks,
best

**Comment by mathmanu** [comment]
There should not be any size different between the original model and the sparse model. In the sparse model, several coefficients are zero - that is all. 

For example check the sizes of these models:

https://github.com/tidsp/caffe-jacinto-models/tree/caffe-0.16/trained/image_segmentation/cityscapes5_jsegnet21v2/initial

https://github.com/tidsp/caffe-jacinto-models/tree/caffe-0.16/trained/image_segmentation/cityscapes5_jsegnet21v2/sparse

Are you using float16 anywhere in your model while training?

**Comment by Wronskia** [comment]
Yes I agree there should not be any difference. here is my training log 
[train.log](https://github.com/tidsp/caffe-jacinto/files/1440500/train.log)

**Comment by Wronskia** [comment]
and the prototxt I am using
and here is the prototxt
[googlenet_deploy.prototxt.docx](https://github.com/tidsp/caffe-jacinto/files/1440529/googlenet_deploy.prototxt.docx)

**Comment by mathmanu** [comment]
Hi,

Your first post said:
**I tried to use a snapshot of a sparsified model from caffe-jacinto and to load it in caffe.**

Your last post said:
**The thing is when I run in caffe-jacinto an inference using this _iter_8000.caffemodel which is half the size of the original, I get the correct results**

Please note that caffe-jacinto is derived from NVIDIA/caffe which has differed from BVLC /caffe quite a bit. I am not sure if NVIDIA/caffe model can be loaded in BVLC/caffe.

What did you mean by caffe in your first post? Is it NVIDIA/caffe of BVLC/caffe?

**Comment by Wronskia** [comment]
Hi,

I meant BVLC/caffe

**Comment by mathmanu** [comment]
Okay. 

Can you try one thing? - train the model in NVIDIA/caffe and check the size. Also see if it can be loaded in BVLC caffe. If not, then you can file an issue in NVIDIA/caffe.

**Comment by Wronskia** [comment]
you are right. The same thing happened in NVcaffe

**Comment by mathmanu** [comment]
Try filing an issue in NVIDIA/caffe. They might be able to help.

**Comment by mathmanu** [comment]
Have you seen this?
https://github.com/NVIDIA/caffe/issues/436

I also noticed the issue that you filed:
https://github.com/NVIDIA/caffe/issues/439

These two seems related.

**Comment by Wronskia** [comment]
Yes thank you manu
